src/compute-client/src/logging.rs

```diff
@@ -22,21 +22,30 @@ use mz_storage::controller::CollectionMetadata;
 include!(concat!(env!("OUT_DIR"), "/mz_compute_client.logging.rs"));
 
 /// Logging configuration.
-#[derive(Arbitrary, Debug, Clone, PartialEq, Serialize, Deserialize)]
+///
+/// An `interval_ns` of 0 specifies that logging is disabled.
+///
```

Comment:
Mentioned earlier, so a bit of a repeat, but I think while this idiom works it is probably a bug. At least, elsewhere in "streaming" what this would imply is an interval width that .. as it approaches zero becomes smaller, and at zero just means "I will leave the timestamps as they happened, rather than round them up to a multiple of anything".

I'm not worried that we are at a point where we need that, but maybe in the clean-up we can make that fix as well. It seems like the `TODO` suggests cleaning something up, and I just wanted to say my bit about intervals going to zeor. :)

Reply:
I agree! What's always in the back of my mind when fixing bugs related to arranged introspection sources is that we want to remove them once the persisted introspection sources are usable. When this time comes, I'd like to make it easy to remove them without having to refactor everything again. In this case what I imagine is that we can leave the protobuf format as is, just remove the `index_logs` field. Introspection is then switched off when `sink_logs` is empty, and the meaning of `interval_ns = 0` is redefined to mean continual logging.

Reply:
Sounds great!